api/src/server/auth.rs

```diff
@@ -0,0 +1,140 @@
+use crate::{
+    error::ResponseError,
+    models::{NewUser, User},
+    schema::users,
+    util::Pool,
+};
+use actix_identity::Identity;
+use actix_web::{get, http, web, HttpResponse, Responder};
+use diesel::{prelude::RunQueryDsl, PgConnection};
+use openid::{Client, DiscoveredClient, Options, Token, Userinfo};
+use reqwest::Url;
+use serde::Deserialize;
+use std::{collections::HashMap, sync::RwLock};
+pub struct Sessions {
+    pub map: HashMap<String, (User, Token, Userinfo)>,
+}
```

Comment:
Not sure if you're planning this, but this should definitely be a DB table, so the locking is much more efficient and it's persistent through restarts etc.

Reply:
Yea ill do that in the follow up